gamestate/user.go

```diff
@@ -45,20 +50,75 @@ func NewUserGateway(gfClient GFClient, loginSocket LoginSocket, gameSocket GameS
 	}
 }
 
+type GfAccountData struct {
+	Email    string
+	Password string
+	Locale   string
+	Name     string
+}
+
 // GameClientGateway provides methods to retrieve information about
 // the real game client.
 type GameClientGateway interface {
 	Version() string
 	Hash() string
 }
 
-func (ug *UserGateway) AuthenticateGFClient(user actions.User, serverLang string) (string, error) {
-	token, accountID, err := ug.gfClient.Authenticate(user, serverLang)
+func (ug *UserGateway) AuthenticateGFClient(jsonAccountPath string, jsonIdentityPath string) (string, error) {
+	content, err := ioutil.ReadFile(jsonAccountPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	identity_manager, err := gfclient_auth.NewIdentityManager(jsonIdentityPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	account_data := new(GfAccountData)
+	json.Unmarshal(content, account_data)
+
+	identity := identity_manager.Get()
+
+	gfclient := gfclient_auth.NewGfClient(
+		identity.Fingerprint.UserAgent,
+		"Chrome/C2.2.23.1813 (49c0acbee1)",
+		identity.Installation_id,
+	)
+
+	err = gfclient.Auth(account_data.Email, account_data.Password, account_data.Locale)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Get game account from user mail
+	game_account_list, err := gfclient.GetGameAccounts()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	// Find game account by its name from account file
+	game_account, err := gfclient_auth.FindGameAccount(account_data.Name, game_account_list)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = gfclient.Iovation(identity_manager, game_account.Id)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	code, err := gfclient.Codes(identity_manager, game_account.Id, game_account.GameId)
```

Comment:
We might want to move the gfclient_auth part to noskit's `gfclient` package which would act as a wrapper around it, what do you think?